### PR TITLE
Feature/6/husky lintstaged

### LINKED ISSUE #6: [FEATURE] Add Husky and appropriate git hook rules
Husky allows [git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) to be run when you commit or push code.

**Describe the solution you'd like**
There are many ways to configure Husky, but at a minimum it should be formatting, linting and running any tests on the code either during the pre-commit or pre-push stages.

An minimal example configuration for `package.json` could be:
```js
  "husky": {
    "hooks": {
      "pre-commit": "yarn run prettier",
      "pre-push": "yarn run lint && yarn run test"
    }
  },
```

I'd be open to any other configurations that are useful though.

**Describe alternatives you've considered**
Husky is the defacto library for this purpose and the developers have experience with it.

### PR DESCRIPTION
**Pull request by julia-rom**
# Description

[Add Husky](https://github.com/globalmangrovewatch/gmw-users/issues/6) & [Lint staged hooks](https://github.com/globalmangrovewatch/gmw-users/issues/7)

Fixes #6, #7

Enable pre and post commit hooks to lint and test code

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Instructions

1. Edit a code file
2. git add
3. git commit
4. see husky and lint staged running in terminal

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- same as described in the instructuions

# Checklist:

- [ X] My code follows the style guidelines of this project
- [X ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mrtt-ui/package.json [hunk 1]
```diff
@@ -47,6 +47,23 @@
...
+      "prettier --write"
+    ],
+    "src/**/*.{js,jsx,tsx,ts}": [
+      "prettier --write",
+      "eslint --fix"
+    ]
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged",
```
Comment (JBurkinshaw) [inline mrtt-ui/package.json:65]: I don't have an answer but I'm wondering if this needs to be here. If it's run when staging occurs, does it need to happen at-pre-commit too? Do you know what best practice is?

**Review by JBurkinshaw — APPROVED** [review]
Code changes look good.

There are a lot of similar sounding commits so when you merge, can you do a squash + merge or rebase + merge and provide a simple description in the commit message? Always good to keep the commit history clean.

File: mrtt-ui/package.json (hunk 1 above)
Comment (julia-rom) [reply mrtt-ui/package.json:65]: I've always used lint-staged with pre-commit checks. This lets each commit be as clean as possible. we may be able to do without the `npm run lint` pre push, although I kind of like it as a safeguard.

**Comment by julia-rom** [comment]
squashed and merged!

**Pull request merged**